### PR TITLE
First Implementation

### PR DESCRIPTION
**Pull request by 256dpi**
@mcollina Ready for a review!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+var net = require('net');
+var tls = require('tls');
+var http = require('http');
+var https = require('https');
+var ws = require('ws');
+var _ = require('underscore');
```
Comment (mcollina) [inline index.js]: I'm kind of against using toolbelts in tiny libraries, like this one, we can do without it :)

File: index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    sharedConfig = {};
+  }
+
+  if(typeof sharedConfig != 'object') {
+    sharedConfig = {};
+  }
+
+  var servers = {};
+
+  _.each(serverConfig, function(config, id){
```
Comment (mcollina) [inline index.js]: you can do without underscore with: 

```
Object.keys(serverConfig).forEach(function(id) { 
  var config = serverConfig
  //...
})
```

File: index.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        }
+      }), callback);
+    },
+    close: function(cb){
+      async.series(_.map(servers, function(server){
+        return server.close.bind(server);
+      }), cb);
+    },
+    destroy: function(cb){
+      async.series(_.map(servers, function(server){
```
Comment (mcollina) [inline index.js]: here you can use `async.mapSeries`

File: index.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    if(typeof config == 'string') {
+      var c = url.parse(config);
+      config = {
+        protocol: c.protocol.replace(/:$/, ''),
+        port: c.port,
+        host: c.hostname
+      };
+    }
+
+    config = _.defaults(config, sharedConfig, {
```
Comment (mcollina) [inline index.js]: Use [xtend](http://npm.im/xtend) to achieve the same result, or just a bunch of ifs :).

File: index.js (hunk 1 above)
Comment (256dpi) [inline index.js]: Yep, I'll remove it!

**Comment by 256dpi** [comment]
Ready to release?

**Comment by mcollina** [comment]
Release! :dancers:

**Pull request merged**